### ISSUE TITLE
KeyError in collect_types prep_args

### ISSUE THREAD

**Issue by elvslv** [dropbox/pyannotate#21]
Annotation of the following method:
```
def method():
    d = {1: {1: 2}}
    return {
        i: {
            (i, k): l
            for k, l in j.iteritems()
        }
        for i, j in d.iteritems()
    }

```
fails with 
```
file.py:25: in test_pyannotate_mocks
    method()
file.py:16: in method
    for i, j in d.iteritems()
file.py:12: in <dictcomp>
    i: {
pyannotate_runtime/collect_types.py:738: in _trace_dispatch
    resolved_types = prep_args(arg_info)
pyannotate_runtime/collect_types.py:475: in prep_args
    resolved_type = resolve_type(arg_info.locals[arg])
E   KeyError: 'j'
```

Not sure if it's a dupe of https://github.com/dropbox/pyannotate/issues/13, but callstack looks pretty similar

**Comment by gvanrossum** [comment]
Aha, this is because dict comprehensions turn into functions with name `<dictcomp>`. We should probably skip any function with a name starting with `'<'`. I'll make a fix.

**Comment by gvanrossum** [comment]
(The fix is just the one suggested in #13; updating the filtering by funcname is a separate job -- I filed #22.)
